### PR TITLE
Add docs for package promotion, releases and branches

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,4 +49,36 @@ This will update the go module files and pull in the most recent vendor files.
...
+As each branch/distribution references its own version, each branch has to be updated with the above. It is encourage to keep the branches in sync related to registry versions.
+
+# Package promotion
+
+A package can go through different stages from snapshot to staging to production. To promote a package from one distribution to an other, it must first be added to the new distribution and as soon as it is added, removed from the old distribution. As an example, a package `foo-1.2.3` is in `snapshot`. Now the content is copied over to `staging` and as soon as the package is merged, the package `foo-1.2.3` should be removed from `snapshot`. If the same package version exists in both branches, the first one is taken. In the above case this is `staging`.
+
+As the staging distribution consists of `production + staging` packages and `snapshot` of `production + staging + snapshot` packages, a package that was promoted to the next stage, is always also available in the previous stage.
+
+The above implies, that as soon as the packages for `staging` as an example are built and the Dockerimage is available, also `snapshot` Dockerimage must be rebuilt as it depends on the `staging` one.
```
Comment (mtojek) [inline README.md:60]: s/Dockerimage/Docker image/g

File: README.md [hunk 2]
```diff
@@ -49,4 +49,36 @@ This will update the go module files and pull in the most recent vendor files.
...
+As soon as the rollout is triggered, it takes up to a minute until all containers are updated and from there, on the CDN side it takes the cache time to update the assets. New packages will be available immediately as they were not served by the CDN previously.
+
+# Branches
+
+The current `package-storage` repository has a few branches. This is a quick summary of the use cases of each branch:
+
+* production: Packages for epr.elastic.co
+* staging: Packages for epr-staging.elastic.co
+* snapshot: Packages for epr-snapshot.elastic.co
+* experiemental: Packages for epr-experimental.elastic.co. These packages are served by Kibana 7.8 and will disappear in the future. No updates should happen to this branch.
```
Comment (mtojek) [inline README.md:83]: nit: experimental

File: README.md (hunk 1 above)
Comment (ruflin) [reply README.md]: fixed, found a a few more.

File: README.md (hunk 2 above)
Comment (ruflin) [reply README.md]: fixed

**Pull request merged**